templates/release.yml

```diff
@@ -0,0 +1,66 @@
+name: Publish Module
+
+on: workflow_dispatch
+  
+jobs:
+  publish-module:
+    name: Tag Release and Publish to Forge
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+        with:
+          ref: ${{ github.ref }}
+          clean: true
+          fetch-depth: 0
+      - name: Get Version
+        id: gv
+        run: |
+          echo "::set-output name=ver::$(jq --raw-output .version metadata.json)"
+      - name: Install the PDK
+        run: |
+          source /etc/os-release
+          wget https://apt.puppet.com/puppet-tools-release-${UBUNTU_CODENAME}.deb
+          sudo dpkg -i puppet-tools-release-${UBUNTU_CODENAME}.deb
+          sudo apt-get update
+          sudo apt-get install pdk
+      - name: Validate Docs
+        run: |
+          documentation=$(pdk bundle exec puppet strings generate --format markdown)
+          if [ $(echo $documentation | grep -Ec "[1-9]+ undocumented|warn") -gt 0 ]; then
+            echo "Please resolve documentation issues detected below:"
+            echo $documentation
+            exit 1
+          fi
+          if output=$(git status --porcelain) && [ ! -z "$output" ]; then
+            echo "The following changes were detected:"
+            git --no-pager diff REFERENCE.md
+            echo "Please submit a PR of changes after running 'pdk bundle exec puppet strings generate --format markdown'"
+            exit 1
+          fi
+      - name: Validate Changelog
+        run: |
+          pdk bundle exec rake changelog
+          grep '### UNCATEGORIZED PRS; LABEL THEM ON GITHUB' CHANGELOG.md
+          if [ $? -eq 0 ]; then
+            echo "Uncategorized PRs found. Please address missing labels in the PRs below:"
+            git --no-pager diff CHANGELOG.md
+            exit 1
+          fi
+          if output=$(git status --porcelain) && [ ! -z "$output" ]; then
+            echo "The following changes were detected:"
+            git --no-pager diff CHANGELOG.md
+            echo "Please address any missing labels and submit a PR of changes after running 'pdk bundle exec rake changelog'"
+            exit 1
+          fi
+      - name: Tag Release
+        uses: ncipollo/release-action@v1
+        with:
+          tag: ${{ steps.gv.outputs.ver }}
+          token: ${{ secrets.GITHUB_TOKEN }}
+          draft: false
+          prerelease: false
+      - name: Build Module
+        run: pdk build
+      - name: Push to Forge
+        run: pdk release publish --forge-token ${{ secrets.FORGE_API_KEY }} --force
```

Comment:
I think these need to explicitly depend on the previous checks so that they don't run unless all validation is green.

Reply:
per verbal chat - disregard, steps run in order 🤦🏻

Reply:
I thought that for a minute too, but looks like all steps within a job depend on previous steps, as opposed to jobs depending on other jobs. I tested this in a private repo, though just to be sure.